### PR TITLE
Support PowerShell 2.0

### PR DESCRIPTION
**Pull request by jberezanski**
This makes the module usable by Chocolatey. Discussed at chocolatey/choco#1201.

### CODE REVIEW

File: src/VSSetup.PowerShell/VSSetup.PowerShell.csproj [hunk 1]
```diff
@@ -41,10 +42,10 @@
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.PowerShell.3.ReferenceAssemblies.1.0.0\lib\net4\System.Management.Automation.dll</HintPath>
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\lib\PowerShell\System.Management.Automation.dll</HintPath>
```
Comment (heaths) [inline src/VSSetup.PowerShell/VSSetup.PowerShell.csproj]: Ick. While you are taking care of the CLA requirement, let me talk with the PowerShell team and see what we can do about this. I'd rather not check in any binaries files, though at least this won't change. Perhaps I can get the asmmeta version (no IL; smaller).

File: test/VSSetup.PowerShell.Test/UtilitiesTests.cs [hunk 2]
```diff
@@ -76,6 +76,42 @@ public void GetAdaptedProperties()
         }
 
         [Fact]
+        public void TryParseVersion_Success()
+        {
+            Version version;
+            var success = Utilities.TryParseVersion("1.2.3", out version);
```
Comment (heaths) [inline test/VSSetup.PowerShell.Test/UtilitiesTests.cs]: Could do all these as a theory,

```csharp
[Theory]
[InlineData("1.2.3", true)]
[InlineData("xxx", false)]
//...
public void TryParseVersion(string versionString, bool expected)
{
  Version version;
  Assert.Equal(expected, Utilities.TryParseVersion(versionString, out version));
  Assert.Equal(new Version(versionString), version);
}
```

File: src/VSSetup.PowerShell/VSSetup.PowerShell.csproj (hunk 1 above)
Comment (jberezanski) [inline src/VSSetup.PowerShell/VSSetup.PowerShell.csproj]: > Ick.

Yeah, but I could not think of any other way of providing that assembly, short of downloading and extracting the PS 2.0 SDK MSI as part of the build process.

> Perhaps I can get the asmmeta version

Would there be any chance of releasing it on NuGet, for the benefit of other projects targeting PS 2.0?

File: test/VSSetup.PowerShell.Test/UtilitiesTests.cs (hunk 2 above)
Comment (jberezanski) [inline test/VSSetup.PowerShell.Test/UtilitiesTests.cs]: Good point, I'll do that.

File: src/VSSetup.PowerShell/VSSetup.PowerShell.csproj (hunk 1 above)
Comment (heaths) [inline src/VSSetup.PowerShell/VSSetup.PowerShell.csproj]: That's what I'm asking the PowerShell team.

**Comment by jberezanski** [comment]
I've reimplemented the tests to use [Theory]. I've also changed them so that the actual `Version.TryParse()` output is used as the expected value.

And here's a funny thing: the behavior of the function is different than its [documentation](https://msdn.microsoft.com/en-us/library/system.version.tryparse(v=vs.110).aspx). When parsing fails, the _result_ out parameter is set to null, not to "a Version object whose major and minor version numbers are 0", as the parameter description claims. This can be confirmed by reviewing [the code](https://referencesource.microsoft.com/#mscorlib/system/version.cs,48d7a76144cbabea).

This is of no consequence to VSSetup, because the value of _result_ is not used if the conversion fails, but for completeness I've fixed `TryParseVersion()` to match the observed behavior instead of the docs.

**Comment by jberezanski** [comment]
Rebased on top of `develop`.

**Comment by jberezanski** [comment]
@heaths Did you get an answer from the PowerShell team?

**Comment by heaths** [comment]
@jberezanski I did and hope to have an update for you very soon.

**Comment by heaths** [comment]
@jberezanski It is now published at https://www.nuget.org/packages/Microsoft.PowerShell.2.ReferenceAssemblies. Please use this instead and rebase to remove the commit containing the binaries before I merge it (so it never ends up in the central git database).

**Comment by jberezanski** [comment]
Excellent! I'll do that soon.

**Comment by jberezanski** [comment]
Done.

**Comment by heaths** [comment]
Thanks. Since the project was downgraded to run on .NET 3.5, I'm getting a couple warnings likely related. Can you clean up the warnings?

> The primary reference "Microsoft.CSharp", which is a framework assembly, could not be resolved in the currently targeted framework. ".NETFramework,Version=v3.5".

Probably just need to remove the reference. And

> Could not read state file "obj\Release\VSSetup.PowerShell.csproj.GenerateResource.Cache". Unable to find assembly 'Microsoft.Build.Tasks.Core, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

**Comment by jberezanski** [comment]
Sorry, I had removed the Microsoft.CSharp references earlier, but they must have reappeared during the last rebase. Fixed now.

I'm not sure about the second warning - it seems like it could be caused by a leftover from a previous compilation using Visual Studio 2015 tools. Could you try a clean build?

**Comment by heaths** [comment]
The second warning went away too after a `git clean -dfx && msbuild /m`.

**Pull request merged**